src/routes/api/db-migrations/db-migration-utils.ts

```diff
@@ -0,0 +1,90 @@
+import { Request } from "express";
+import fs from "fs";
+import path from "path";
+import { isNodeDev, isNodeTest, isNodeProd, getNodeEnv } from "utils/is-node-env";
+import { getLogger } from "config/logger";
+import { exec as execOrigin } from "child_process";
+import { promisify } from "util";
+
+const exec = promisify(execOrigin);
+const logger = getLogger("DBMigration");
+
+/**
+ * Return the common param in req.body `targetScript` to migrate up or rollback
+ * Note: Intentially to make this param case sensitive to be more safe
+ */
+export const getTargetScript = (req: Request) => {
+	let targetScript = (req.body || {}).targetScript;
+	if (!targetScript) {
+		throw {
+			statusCode: 400,
+			message: `"targetScript" is mandatory in the request body, but found none.`
+		};
+	}
+	//just make sure the script name matching the ones in db down the track.
+	if (!targetScript.endsWith(".js")) targetScript = targetScript + ".js";
+	return targetScript;
+};
+
+/**
+ * Make sure the script to migrate up or rollback is the lastest scripts in this build.
+ * So following scenarios will failed.
+ * Person A merge script 1, person B merge scripts 2. Now we CANNOT migrate anymore. Need to revert one PR first inorder to migrate.
+ *
+ * In short, encourage ppl to over-communicate when it comes to db migrations.
```

Comment:
/nit I'd've double-checked English in these comments 😛

Reply:
👍 thanks mate, just double checked and fixed things.